libs/designsystem/src/lib/components/list/list-swipe-action.type.ts

```diff
@@ -1,14 +1,23 @@
-export type ListSwipeActionType = 'success' | 'warning' | 'danger';
+import {
+  ItemSlidingSide as ListSwipeDirection,
+  ItemSwipeAction,
+} from '../item-sliding/item-sliding.types';
 
-export type ListSwipeDirection = 'right' | 'left';
-
-export type ListSwipeEnd = 'start' | 'end';
-
-export interface ListSwipeAction {
+// Copy each property in ItemSwipeAction but make it possible to provide them via a function.
+type ListSwipeActionBaseTypes = Pick<ItemSwipeAction, 'title' | 'icon' | 'type' | 'isDisabled'>;
+export type ListSwipeAction = {
+  [Property in keyof ListSwipeActionBaseTypes]:
+    | ((item: any) => ListSwipeActionBaseTypes[Property])
+    | ListSwipeActionBaseTypes[Property];
+} & {
```

Comment:
None of the tests for the template driven list failed and the cookbook was able to build. So i took that as a sign that nothing was broken. But it could be a good idea to test it out in the cookbook.